homeassistant/components/weather/yandex.py

```diff
@@ -0,0 +1,218 @@
+
+# API limit is 50 req/day
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=40)
+
+CONDITION_TO_HA = {
+    "clear": "clear-night",
+    "partly-cloudy": "partlycloudy",
+    "cloudy": "cloudy",
+    "overcast": "cloudy",
+    "partly-cloudy-and-light-rain": "rainy",
+    "partly-cloudy-and-rain": "pouring",
+    "overcast-and-rain": "rainy",
+    "overcast-thunderstorms-with-rain": "lightning-rainy",
+    "cloudy-and-light-rain": "rainy",
+    "overcast-and-light-rain": "rainy",
+    "cloudy-and-rain": "rainy",
+    "overcast-and-wet-snow": "snowy-rainy",
+    "partly-cloudy-and-light-snow": "snowy",
+    "partly-cloudy-and-snow": "snowy",
+    "overcast-and-snow": "snowy",
+    "cloudy-and-light-snow": "snowy",
+    "overcast-and-light-snow": "snowy",
+    "cloudy-and-snow": "snowy"
+}
+
+DAYPARTS_TIMEDELTA = (
+    ("night", timedelta(hours=0)),
+    ("morning", timedelta(hours=6)),
+    ("day", timedelta(hours=12)),
+    ("evening", timedelta(hours=18))
+)
+
+
+CONF_RATE = "rate"
+CONF_UPDATE_TIMEOUT = "update_timeout"
+RATE_INFORMERS = "informers"
+RATE_FORECAST = "forecast"
+VALID_RATES = (RATE_INFORMERS, RATE_FORECAST)
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Optional(CONF_RATE): vol.Any(*VALID_RATES),
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_LATITUDE): cv.latitude,
+    vol.Optional(CONF_LONGITUDE): cv.longitude,
+    vol.Optional(CONF_UPDATE_TIMEOUT): cv.positive_timedelta
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the yandex weather platform."""
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+    if None in (latitude, longitude):
+        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
+        return
+    latitude = float(latitude)
+    longitude = float(longitude)
+    websession = async_get_clientsession(hass)
+
+    _LOGGER.debug("Initializing for coordinates %s, %s", latitude, longitude)
+
+    async_add_entities([YandexWeather(
+        websession, latitude, longitude, config)], True)
+
+
+def not_none(prop):
+    """A decorator for a class method to use a non-None value."""
+    def __inner(func):
+        @wraps(func)
+        def decore(self, *args, **kwargs):
+            if getattr(self, prop) is not None:
+                return func(self, *args, **kwargs)
+        return decore
+    return __inner
+
+
+class YandexWeather(WeatherEntity):
+    """Representation of a weather condition."""
+
+    def __init__(self, session, lat, lon, config):
+        """Initialise the platform with a data instance and station name."""
+        self._station_name = config.get(CONF_NAME)
+        if self._station_name is None:
+            self._station_name = "Weather at {}, {}".format(lat, lon)
+        self._api_key = config.get(CONF_API_KEY)
```

Comment:
The client object should be created during `async_setup_platform()` and passed to `YandexWeather()`. Also, you should check that the API key is valid and fail the platform setup if not. Otherwise will the users end-up with a non-functional platform in their setup.